distill.py

```diff
@@ -0,0 +1,218 @@
+"""
+Distillation Script for Wake Vision and Visual Wake Words Datasets
+"""
+import numpy as np
+import os
+
+os.environ["KERAS_BACKEND"] = "jax"
+
+# Note that keras should only be imported after the backend
+# has been configured. The backend cannot be changed once the
+# package is imported.
+import keras
+
+import tensorflow as tf
+import tensorflow_datasets as tfds
+import numpy as np
+
+from experiment_config import default_cfg, get_cfg
+from wake_vision_loader import get_wake_vision, get_miaps
+from vww_loader import get_vww
+
+import wandb
+from wandb.keras import WandbMetricsLogger
+
+
+def distill(teacher_config, student_cfg=default_cfg, extra_evals=["distance_eval", "miap_eval"]):
+    wandb.init(project="wake-vision", name=student_cfg.EXPERIMENT_NAME+"_Distill", config=student_cfg)
+
+    if student_cfg.TARGET_DS == "vww":
+        train, val, test = get_vww(student_cfg)
+    else:
+        train, val, test = get_wake_vision(student_cfg)
+
+    student = keras.applications.MobileNetV2(
+        input_shape=student_cfg.INPUT_SHAPE,
+        alpha=student_cfg.MODEL_SIZE,
+        weights=None,
+        classes=student_cfg.NUM_CLASSES,
+    )
+
+    teacher_path = teacher_config.SAVE_FILE
+    print("Loading Teacher:"
+          f"{teacher_path}")
+    teacher = keras.saving.load_model(teacher_path)
+    teacher.trainable = False
+    
+    print("Student Summary:")
+    student.summary()
+
+    print("Teacher Summary:")
+    teacher.summary()
+
+    print("Teacher Evaluation:")
+    print(teacher.evaluate(val, verbose=1))
+
+    class Distiller(keras.Model):
+        def __init__(self, student, **kwargs):
+            super().__init__(**kwargs)
+            self.student = student
+
+        def compile(
+            self,
+            optimizer,
+            metrics,
+            student_loss_fn,
+            distillation_loss_fn,
+            alpha=1.0,
+            **kwargs,
+        ):
+            """Configure the distiller.
+
+            Args:
+                optimizer: Keras optimizer for the student weights
+                metrics: Keras metrics for evaluation
+                student_loss_fn: Loss function of difference between student
+                    predictions and ground-truth
+                distillation_loss_fn: Loss function of difference between soft
+                    student predictions and soft teacher predictions
+                alpha: weight to student_loss_fn and 1-alpha to distillation_loss_fn
+                temperature: Temperature for softening probability distributions.
+                    Larger temperature gives softer distributions.
```

Comment:
It seems that these arg descriptions are not completely corresponding to the function arguments